### ISSUE TITLE
ArrayField + Datagrid: if embedded data doesn't have an `id`, rows won't have a `key` and will cause a React warning

### ISSUE THREAD

**Issue by nselikoff** [marmelab/react-admin#7541]
**What you were expecting:**
Using an ArrayField + Datagrid for embedded data that doesn't have an `id` prop should work without console errors

**What happened instead:**
I get this error in the console: `Warning: Each child in a list should have a unique "key" prop.`

**Steps to reproduce:**
Given this Show component for an "album" record:

```jsx
export const AlbumShow = () => (
  <Show>
    <SimpleShowLayout>
      <ArrayField source="features">
        <Datagrid>
          <TextField source="name" />
          <TextField source="value" />
        </Datagrid>
      </ArrayField>
    </SimpleShowLayout>
  </Show>
);
```

Where an album has embedded features that look like this (note that `features` don't have an `id` prop):
```json
{
  "id": 0,
  "features": [
    { "name": "title", "value": "Test Album" }
  ]
}
```

**Related code:**
You can see repro this in the [simple example](https://github.com/marmelab/react-admin/tree/master/examples/simple) (checked out and run locally; I don't see warnings in codesandbox's console)
Go to a post show page that has backlinks, e.g. http://localhost:8080/#/posts/12/show

**Other information:**
This is where the key is set based on the id, which doesn't exist in this case: https://github.com/marmelab/react-admin/blob/a408de0db2c6db0281b3d39e27c1f2de3deed44e/packages/ra-ui-materialui/src/list/datagrid/DatagridBody.tsx#L48

**Environment**

* React-admin version: 4.0.0
* Last version that did not exhibit the issue (if applicable):
* React version: 18.0.0
* Browser: Chrome
* Stack trace (in case of a JS error):

**Comment by WiXSL** [comment]
Reproduced. Thanks!

**Comment by WiXSL** [comment]
Note that in version 4, you have to pass `bulkActionButtons={false}` in a `Datagrid` inside an `ArrayField`.
This is clarified in the [Upgrade guide](https://marmelab.com/react-admin/Upgrade.html#datagrid-hasbulkactions-prop-has-been-removed).

**Comment by WiXSL** [comment]
Fixed by #7548

**Comment by nselikoff** [comment]
Thanks for the quick fix!
